packages/jellyfish-api-core/__tests__/category/loan/updateLoanToken.test.ts

```diff
@@ -0,0 +1,289 @@
+      [loanTokenId]: {
+        token: {
+          1: {
+            symbol: 'Token2',
+            symbolKey: 'Token2',
+            name: 'Token1',
+            decimal: 8,
+            limit: 0,
+            mintable: true,
+            tradeable: true,
+            isDAT: true,
+            isLPS: false,
+            finalized: false,
+            isLoanToken: true,
+            minted: 0,
+            creationTx: loanTokenId,
+            creationHeight: await testing.container.getBlockCount() - 1,
+            destructionTx: '0000000000000000000000000000000000000000000000000000000000000000',
+            destructionHeight: -1,
+            collateralAddress: expect.any(String)
+          }
+        },
+        priceFeedId,
+        interest: 0.01
+      }
+    })
+  })
+
+  it('should updateLoanToken by id as token', async () => {
+    await testing.rpc.loan.updateLoanToken('1', { symbol: 'Token2' }) // Update by id
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    expect(data[loanTokenId].token[1].symbol).toStrictEqual('Token2')
+  })
+
+  it('should updateLoanToken by creationTx as token', async () => {
+    await testing.rpc.loan.updateLoanToken(loanTokenId, { symbol: 'Token2' }) // Update by creationTx
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    expect(data[loanTokenId].token[1].symbol).toStrictEqual('Token2')
+  })
+
+  it('should not updateLoanToken if token does not exist', async () => {
+    const promise = testing.rpc.loan.updateLoanToken('Token2', { symbol: 'Token2' }) // Does not exist
+    await expect(promise).rejects.toThrow('RpcApiError: \'Token Token2 does not exist!\', code: -8, method: updateloantoken')
+  })
+
+  it('should updateLoanToken if symbol is more than 8 letters', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token3',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const loanTokenId = await testing.container.call('setloantoken', [{
+      symbol: 'Token3',
+      priceFeedId
+    }, []])
+    await testing.generate(1)
+
+    await testing.rpc.loan.updateLoanToken('Token3', { symbol: 'x'.repeat(9) }) // 9 letters
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    const index = Object.keys(data).indexOf(loanTokenId) + 1
+    expect(data[loanTokenId].token[index].symbol).toStrictEqual('x'.repeat(8)) // Only remain the first 8 letters
+  })
+
+  it('should updateLoanToken by given symbol', async () => {
+    await testing.rpc.loan.updateLoanToken('Token1', { symbol: 'Token2' })
+    await testing.generate(1)
+
+    const data = await testing.container.call('listloantokens', [])
+    const index = Object.keys(data).indexOf(loanTokenId) + 1
+    expect(data[loanTokenId].token[index].symbol).toStrictEqual('Token2')
+  })
```

Comment:
duplicate with the test starting from LN:59 ?

Reply:
as I can see `should updateLoanToken by symbol as token` and `should updateLoanToken by given symbol` both does the same thing. i.e. `Update by symbol`. the only difference is the check/expect
Have a look again? if you think it's not duplicate, you can leave it.

Reply:
Okay you are right. I thought you meant other thing